### PR TITLE
Make the overlay work correctly and use it everywhere

### PR DESCRIPTION
**Pull request by K900**
This basically refactors the entire structure of the overlay and modules, so instead of packages from the flake, the overlay with all the packages is applied to the user's nixpkgs and then the packages from the final nixpkgs are used in the modules. The packages built against pinned nixpkgs are still exposed through the flake.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -12,48 +12,32 @@ rec {
...
+      allLaunchers = pkgs.symlinkJoin {
         name = "allLaunchers";
-        paths = builtins.attrValues final;
+        paths = builtins.filter pkgs.lib.isDerivation (builtins.attrValues overlay);
       };
-      # return this package set with a given nixpkgs instance
-      withNixpkgs = p: packages p;
+
+      unwrapped = pkgs.anime-games-launcher-unwrapped;
+      regular = pkgs.anime-games-launcher;
```
Comment (ezKEa) [inline flake.nix:35]: These can probably be removed. Very very old backwards compatibility thing.

File: module/default.nix [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+    ./borb.nix
+    ./hrl.nix
+    ./hl.nix
+    ./waves.nix
+    ./sleepy.nix
     ./hosts.nix
     ./version.nix
   ];
+
+  nixpkgs.overlays = [ overlay ];
```
[the remark this answers is not in the document]
Comment (K900) [reply module/default.nix:14]: The overlay shouldn't get in the way of that though? You can just set `package = inputs.aagl.packages.${system}.whatever-launcher` explicitly.

**Comment by K900** [comment]
Realized we don't actually need to pass the overlay from the flake at all, changed that.

**Comment by ezKEa** [comment]
Appreciate the support! :pray:

**Pull request merged**